appserver/common/amx-javaee/src/main/java/org/glassfish/admin/amx/impl/j2ee/RegistrationSupport.java

```diff
@@ -311,25 +311,27 @@ else if (ejbType.equals("Message-driven"))
         }
         else if (ejbType.equals("Session"))
         {
-            if (ejbSessionType.equals("Stateless"))
+            if ("Stateless".equals(ejbSessionType))
             {
                 intf = StatelessSessionBean.class;
                 impl = StatelessSessionBeanImpl.class;
             }
-            else if (ejbSessionType.equals("Stateful"))
+            else if ("Stateful".equals(ejbSessionType))
             {
                 intf = StatefulSessionBean.class;
                 impl = StatefulSessionBeanImpl.class;
             }
-            else if (ejbSessionType.equals("Singleton")) // EJB 3.1
+            else if ("Singleton".equals(ejbSessionType)) // EJB 3.1
             {
                 intf = SingletonSessionBean.class;
                 impl = SingletonSessionBeanImpl.class;
             }
             else
             {
-                throw new IllegalArgumentException( "Unknown ejbSessionType: " + ejbSessionType + ", expected Stateless or Stateful");
+                throw new IllegalArgumentException("Unknown ejbSessionType: " + ejbSessionType + ", expected Stateless or Stateful");
             }
+        } else {
+            throw new IllegalArgumentException("Unknown ejbType: " + ejbType + ", expected Entity, Message-driven or Session");
```

Comment:
This case needs to be added in order to prevent that `impl` and `intf` cannot be null when calling `registerJ2EEChild` below which requires them to be non-null. Therefore I opted to also throw an `IllegalArgumentException` for ùnexpected `ejbType` similar to the one present for unexpected `ejbSessionType`.